test_http_server.py

```diff
@@ -1,20 +1,26 @@
 # -*- coding:UTF-8 -*-
 # from __future__ import unicode_literals
-from server import response_error, response_ok, respond, parse_request, resolve_uri
+from server import response_error, response_ok, parse_request, resolve_uri
+from concurrent_server import create_server
 import time
 
 import pytest
 import socket
 
 
-@pytest.yield_fixture
-def server_process(scope='session'):
+@pytest.fixture(scope='session')
+def server_process(request):
     from multiprocessing import Process
-    process = Process(target=respond)
+    process = Process(target=create_server)
     process.daemon = True
     process.start()
-    time.sleep(.01)
-    yield process
+    time.sleep(.02)
+
+    def cleanup():
+        process.terminate()
+
+    request.addfinalizer(cleanup)
+    return process
```

Comment:
nice.  baddass concurrent tests.  Might be fun to add a test that spins off like 200 separate processes and makes a series of requests from each.  Load testing FTW!